DirectX11/Hunting.cpp

```diff
@@ -46,7 +43,7 @@ static void DumpUsageResourceInfo(HANDLE f, std::set<uint32_t> *hashes, char *ta
 	DWORD written; // Really? A >required< "optional" paramter that we don't care about?
 	bool nl;
 
-	for (orig_hash = hashes->begin(); orig_hash != hashes->end(); orig_hash++) {
+	for (auto orig_hash = hashes->begin(); orig_hash != hashes->end(); orig_hash++) {
```

Comment:
This is a coding style change, and debatable whether it is better or worse. "auto" saves a lot of keystrokes, but can mask bugs where an incorrect type has been used. I won't object to new code using "auto", but there needs to be a good reason for altering already written and well tested code.

Reply:
We don't have an official coding style for 3DMigoto, so for your own code you are fairly free to use whatever you prefer, provided that it is readable and doesn't mask bugs. There might be a few things here or there we pick up on, but until we adopt an actual coding style, please consider them suggestions rather than rules. However, if you are working in existing code, please refrain from chainging the coding style unecessarily and try to adopt whatever is already there.

This discussion isn't one way - if you feel strongly that a particular coding style is good/bad/ugly you can bring it up for discussion. If we were to adopt a coding style, Bo3b has suggested Google's coding style guidelines (but again, no decision has been made to actually do so): http://google.github.io/styleguide/

Any functions you see that declare all their variables at the top like this one are most likely written by myself following the coding styles of the Linux Kernel, which make functions more readable than declaring variables inline. This is definitely not a coding style we are considering as a whole for 3DMigoto, as it is designed for a C project, and some of the guidelines do not work well in C++, but (like all coding style guidelines) it does make some good points.